cmd/opm/alpha/veneer/composite.go

```diff
@@ -0,0 +1,173 @@
+					continue
+				}
+
+				if _, ok := catalogBuilderMap[catalog.Name]; !ok {
+					builderMap := make(composite.BuilderMap)
+					for _, schema := range catalog.Builders {
+						builder, err := builderForSchema(schema, composite.BuilderConfig{
+							ContainerCfg: composite.ContainerConfig{
+								ContainerTool: containerTool,
+								BaseImage:     catalog.Destination.BaseImage,
+								WorkingDir:    catalog.Destination.WorkingDir,
+							},
+							OutputType:       output,
+							CurrentDirectory: wd,
+						})
+						if err != nil {
+							log.Fatalf("getting builder %q for catalog %q: %s", schema, catalog.Name, err)
+						}
+						builderMap[schema] = builder
+					}
+					catalogBuilderMap[catalog.Name] = builderMap
+				}
+			}
+
+			// if there were errors validating the catalog configuration then exit
+			if setupFailed {
+				//build the error message
+				var errMsg string
+				for cat, errs := range setupErrors {
+					errMsg += fmt.Sprintf("\nCatalog %s:\n", cat)
+					for _, err := range errs {
+						errMsg += fmt.Sprintf("  - %s\n", err)
+					}
+				}
+				log.Fatalf("catalog configuration file field validation failed: %s", errMsg)
+			}
+
+			veneer.CatalogBuilders = catalogBuilderMap
+
+			compositeData, err := os.Open(compositeFile)
+			if err != nil {
+				log.Fatalf("opening composite config file %q: %s", compositeFile, err)
+			}
+			defer compositeData.Close()
+
+			// parse data to composite config
+			compositeConfig := &composite.CompositeConfig{}
+			compositeDoc := json.RawMessage{}
+			compositeDecoder := yaml.NewYAMLOrJSONDecoder(compositeData, 4096)
+			err = compositeDecoder.Decode(&compositeDoc)
+			if err != nil {
+				log.Fatalf("decoding composite config: %s", err)
+			}
+			err = json.Unmarshal(compositeDoc, compositeConfig)
+			if err != nil {
+				log.Fatalf("unmarshalling composite config: %s", err)
+			}
+
+			if compositeConfig.Schema != composite.CompositeSchema {
+				log.Fatalf("%q has unknown schema, should be %q", compositeFile, composite.CompositeSchema)
+			}
+
+			err = veneer.Render(cmd.Context(), compositeConfig, validate)
+			if err != nil {
+				log.Fatalf("rendering the composite veneer: %s", err)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&output, "output", "o", "json", "Output format (json|yaml)")
+	// TODO: Investigate ways to do this without using a cli tool like docker/podman
+	// cmd.Flags().StringVar(&containerTool, "container-tool", "docker", "container tool to be used when rendering veneers (should be an equivalent replacement to docker - similar to podman)")
+	cmd.Flags().BoolVar(&validate, "validate", true, "whether or not the created FBC should be validated (i.e 'opm validate')")
+	cmd.Flags().StringVarP(&compositeFile, "composite-config", "c", "catalog/config.yaml", "File to use as the composite configuration file")
+	cmd.Flags().StringVarP(&catalogFile, "catalog-config", "f", "catalogs.yaml", "File to use as the catalog configuration file")
+	return cmd
```

Comment:
could we add a flag to account for a previous catalog generation so we dont have to pull all images again? or an overwrite option for the output in case it's non empty so we can warn users if they make a veneer change that would strand users on a particular version for example?

Reply:
Are you thinking about the use-case where we want to make incremental updates only to an existing catalog?  According to the document (https://docs.google.com/document/d/1uEZdVA06GGahPl3ygnGrFLRkrohaQjs25mVcXToYakE/edit#heading=h.kjc5gjiy2pkd) SOP will be to overwrite the specific contribution in all cases. 

I'd be interested in talking more about the use-case, but def not barring this PR merge.